### PR TITLE
Added user authorization in Cosv

### PR DESCRIPTION
**Pull request by Cheshiriks**

### CODE REVIEW

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/configs/WebConfig.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.boot.web.reactive.function.client.WebClientCustomizer
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.codec.json.Jackson2JsonDecoder
+import org.springframework.http.codec.json.Jackson2JsonEncoder
+
+@Configuration
+class WebConfig {
```
Comment (akuleshov7) [inline save-cosv/src/main/kotlin/com/saveourtool/save/cosv/configs/WebConfig.kt:18]: Wtf is that? Why is it here?

Comment (akuleshov7) [reply save-cosv/src/main/kotlin/com/saveourtool/save/cosv/configs/WebConfig.kt:18]: Need to make it common

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/storage/AvatarKey.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.saveourtool.save.cosv.storage
+
+import com.saveourtool.save.utils.AvatarType
+
+/**
+ * @property type
+ * @property objectName
+ */
+data class AvatarKey(
```
Comment (akuleshov7) [inline save-cosv/src/main/kotlin/com/saveourtool/save/cosv/storage/AvatarKey.kt:9]: move to common please

**Review by akuleshov7 — APPROVED** [review]

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/repositorysave/GitRepository.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.saveourtool.save.entities.Git
+import com.saveourtool.save.entities.Organization
+import com.saveourtool.save.spring.repository.BaseEntityRepository
+import org.springframework.stereotype.Repository
+
+/**
+ * Repository of git
+ */
+@Repository
+interface GitRepository : BaseEntityRepository<Git> {
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-cosv/src/main/kotlin/com/saveourtool/save/cosv/repositorysave/GitRepository.kt:12]: This is used in organization organization and we can go to this view from COSV

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/repositorysave/ProjectRepository.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor
+import org.springframework.data.repository.query.QueryByExampleExecutor
+import org.springframework.stereotype.Repository
+
+/**
+ * The repository of project entities
+ */
+@Repository
+interface ProjectRepository : JpaRepository<Project, Long>, QueryByExampleExecutor<Project>,
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-cosv/src/main/kotlin/com/saveourtool/save/cosv/repositorysave/ProjectRepository.kt:15]: We have a list of tools in the organization, so we had to add projects

File: save-cosv/src/main/kotlin/com/saveourtool/save/cosv/configs/WebConfig.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-cosv/src/main/kotlin/com/saveourtool/save/cosv/configs/WebConfig.kt:18]: Added to issue

**Pull request merged**